hips/hip-0019.md

```diff
@@ -0,0 +1,186 @@
+  version: 11.4.1
+- name: common
+  repository: https://charts.bitnami.com/bitnami
+  version: 2.2.2
+digest: sha256:5a55c2b684433bf6fa9efcf0d3c4ab23ceef313808203b766c485b6e03bda60f
+generated: "2022-12-13T02:28:18.307577267Z"
+images:
+  - name: wordpress
+    image: docker.io/bitnami/wordpress:6.0.2-debian-11-r9
+    digests: 
+      - digest: sha256:b4cb055a643d1d51d1678f3e011cf008b227852bc0a4118126deeed844ccfb6a
+        arch: linux/amd64
+      - digest: sha256:4d2760961bdd2e9956d752d637e7795e2eea973c25825bd18cb881c3edeee8ff
+        arch: linux/arm64
+    chart: wordpress
+  - name: bitnami-shell
+    image: docker.io/bitnami/bitnami-shell:11-debian-11-r37
+    digests: 
+      - digest: sha256:0946f1d16c010bed0a47e3c09ff193d898c868044bb4c1b037c83b7c53c26deb
+        arch: linux/amd64
+    chart: wordpress
+  - name: apache-exporter
+    image: docker.io/bitnami/apache-exporter:0.11.0-debian-11-r42
+    digests: 
+      - digest: sha256:ea4aa97834500297e4252f5f98492c52a7bfd1993bd5ef4e485ae84996b1c63a
+        arch: linux/amd64
+    chart: wordpress
+```
+
+## Backwards compatibility
+
+There isn't any backward incompatibility, since there's no modification of any existing schema. Charts making use of the `annotations` and `images-lock` file would benefit of the feature, others just wouldn't.
+
+## Security implications
+
+Initially, there are no secutiry implications.
+
+## How to teach this
+
+Firstly, users will need to know how to define the `annotations` so the `images-lock.yaml` file is properly generated. This could be probably addressed through a section either in the `Chart.yaml` [documentation](https://helm.sh/docs/topics/charts/#the-chartyaml-file) or next to it. 
+
+Then, a description about the lock file would be needed. This is, to ensure that users understand the benefits they can get from it: having a bill of materials of the chart, overriding images at installation, migrating whole charts...
+
+Once users have defined the `annotations` properly and understand what's the `images-lock.yaml` file for, they'll need to know how to use features built on top of it. These may come in different ways: new commands (i.e. `migrate` for moving Charts between registries alongside their images, plus rewriting the lock file), parameters (i.e. `push --with-images` could do the same) or just transparently (i.e. `helm install` plus a modified lock file would pin specific images). Depending on this, the affected part of the CLI documentation should be updated accordingly.
+
+It may be less relevant to the end users how is the `images-lock.yaml` packaged alongside the Helm Chart, whether if it is pushed to a different OCI blob, or how is it named in the registry. These points should be mentioned somewhere as well, but it's likely that with less emphasis. 
+
+In addition to the points made above, given the size of the improvement, promotion through conference talks, presentations or any other public conversation would be strongly suggested.
```

Comment:
Maybe something in here about the linter and [chart-testing](https://github.com/helm/chart-testing)